### PR TITLE
kie-issues#1250: prepare for branch,add gpg and svn configuration

### PR DESCRIPTION
**Pull request by jstastny-cz**
Addresses:
* apache/incubator-kie-issues#1250
* apache/incubator-kie-issues#1251

* Adding release configuration entries for gpg and svn into branch.yaml config. (to drools and optaplanner to come later).
* removing suffix from nightly image name (-nightly) and tag (-yyyymmdd)
* removing operator from the pipeline for now
* removing images promoting
* adding disable: images-deploy configuration option to be set to `true` on release branch.
* Adding release pipeline shared library var with gpg and svn methods

### CODE REVIEW

File: jenkins-pipeline-shared-libraries/vars/release.groovy [hunk 1]
```diff
@@ -0,0 +1,28 @@
+def gpgImportKeyFromFileWithPassword(String gpgKeyCredentialsId, String gpgKeyPasswordCredentialsId) {
+    withCredentials([file(credentialsId: gpgKeyCredentialsId, variable: 'SIGNING_KEY')]) {
+        withCredentials([file(credentialsId: gpgKeyPasswordCredentialsId, variable: 'SIGNING_KEY_PASSWORD')]) {
```
Comment (jstastny-cz) [inline jenkins-pipeline-shared-libraries/vars/release.groovy]: these won't be file

**Comment by rodrigonull** [comment]
@jstastny-cz Can you please also update this file: https://github.com/apache/incubator-kie-kogito-pipelines/blob/main/.ci/jenkins/Jenkinsfile.weekly.cloud#L211 to remove the nightly suffix?

File: jenkins-pipeline-shared-libraries/vars/release.groovy [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    }
+}
+
+boolean gpgIsValidDetachedSignature(String file, String signature) {
+    return sh(returnStatus: true, script: "gpg --batch --verify ${signature} ${file}") == 0
+}
+
+def svnUploadFileToRepository(String svnRepository, String svnCredentialsId, String... files) {
+    throw new NotImplementedException("stub");
+}
```
Comment (rodrigonull) [inline jenkins-pipeline-shared-libraries/vars/release.groovy]:
```suggestion
def svnUploadFileToRepository(String svnRepository, String svnCredentialsId, String releaseVersion, String... files) {
    withCredentials([usernamePassword(credentialsId: svnCredentialsId, usernameVariable: 'ASF_USERNAME', passwordVariable: 'ASF_PASSWORD')]) {
        sh "svn co --depth=empty ${svnRepository} svn-kie"
        for (file in files) {
            sh "cp ${file} svn-kie/${releaseVersion}/"
        }
        sh """
        svn add "svn-kie/${releaseVersion}"
        cd svn-kie
        svn ci --non-interactive --no-auth-cache --username ${ASF_USERNAME} --password '${ASF_PASSWORD}' -m "Apache KIE ${releaseVersion} artifacts"
        rm -rf svn-kie
        """
    }
}
```

**Review by rodrigonull — CHANGES_REQUESTED** [review]

**Review by porcelli — APPROVED** [review]

**Review by rodrigonull — APPROVED** [review]
LGTM.

**Pull request merged**